sound/soc/intel/Kconfig

```diff
@@ -77,7 +77,7 @@ config SND_SOC_ACPI_INTEL_MATCH
 
 config SND_SOC_ACPI_INTEL_SDCA_QUIRKS
 	tristate
-	imply SND_SOC_SDCA
+	select SND_SOC_SDCA
```

Comment:
> Can you also add:
> 
> Reported-by: kernel test robot [lkp@intel.com](mailto:lkp@intel.com) Closes: https://lore.kernel.org/oe-kbuild-all/202411021722.DiX1Y5sf-lkp@intel.com/

Of course, thanks for reminding me.

Reply:
@ujfalusi @bardliao This makes me think SDCA library should be default=y for Intel in the near term. i.e. we should plan this soon.

Reply:
I believe if you change this to a select you also need to pull forward the depends, since select will ignore the depends of its target. So you probably need to add a depends on ACPI into SND_SOC_ACPI_INTEL_SDCA_QUIRKS.

Reply:
and to add, the advice is that the select should not be used for user visible options...

SND_SOC_ACPI_INTEL_MATCH is only selected if ACPI is enabled, but randconfig might be able to generate a config which can fail.

Reply:
might be this?
```diff
diff --git a/sound/soc/intel/Kconfig b/sound/soc/intel/Kconfig
index 14461dee3e52..46b45f390ae9 100644
--- a/sound/soc/intel/Kconfig
+++ b/sound/soc/intel/Kconfig
@@ -70,14 +70,15 @@ if SND_SOC_INTEL_SST_TOPLEVEL || SND_SOC_SOF_INTEL_TOPLEVEL
 
 config SND_SOC_ACPI_INTEL_MATCH
 	tristate
-	select SND_SOC_ACPI if ACPI
+	depends on ACPI
+	select SND_SOC_ACPI
 	select SND_SOC_ACPI_INTEL_SDCA_QUIRKS
 	# this option controls the compilation of ACPI matching tables and
 	# helpers and is not meant to be selected by the user.
 
 config SND_SOC_ACPI_INTEL_SDCA_QUIRKS
 	tristate
-	imply SND_SOC_SDCA
+	select SND_SOC_SDCA
 
 endif ## SND_SOC_INTEL_SST_TOPLEVEL || SND_SOC_SOF_INTEL_TOPLEVEL
 ```

Reply:
Sounds good to me. I don't know how can SND_SOC_ACPI_INTEL_MATCH work without ACPI. Also, I think we can make SND_SOC_SDCA invisible in this PR.